elyra/pipeline/component_registry.py

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import time
+
+from jinja2 import Environment, PackageLoader
+from elyra.pipeline.component import ComponentParser, Component
+from traitlets.config import LoggingConfigurable
+from types import SimpleNamespace
+from typing import List, Dict
+
+
+class ComponentRegistry(LoggingConfigurable):
+    """
+    Component Registry, responsible to provide a list of available components
+    for each runtime. The registry uses component parser to read and parse each
+    component entry from the catalog and transform them into a component value object.
+    """
+
+    _generic_components: Dict[str, Component] = {
+        "notebooks": Component(id="notebooks",
```

Comment:
Nit: key and id should be singular for consistency

Reply:
Discussed this, but this would imply migration of old pipelines

Reply:
Add here: https://github.com/elyra-ai/elyra/issues/1773